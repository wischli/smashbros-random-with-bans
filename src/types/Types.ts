export interface IChar {
  id: number;
  name: string;
  enabled: boolean;
  played: boolean;
  display: boolean;
  echo: number[];
  media: string;
}

export interface ICState {
  enabled: number[];
  played: number[];
  disabled: number[];
  hidden: number[];
  date: string;
}

export interface IState {
  enabled: number[];
  played: number[];
  disabled: number[];
  hidden: number[];
}

export interface ICookies {
  characters: ICState | boolean;
  notice: boolean;
}
