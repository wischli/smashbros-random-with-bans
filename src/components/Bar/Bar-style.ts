import { color, themeButtonStyle } from '../../layout/themeStyle'

export const buttonStyle: React.CSSProperties = {
  ...themeButtonStyle,
  width: '100%',
  backgroundColor: color.bgNavSecBtn,
}

export const navStyle: React.CSSProperties = {
  width: 'calc(100% - 20px)',
  padding: '10px',
  display: 'flex',
  flexWrap: 'nowrap',
  justifyContent: 'space-between',
  backgroundColor: color.nav,
  position: 'fixed',
  height: 50,
  top: 0,
  alignItems: 'center',
}
