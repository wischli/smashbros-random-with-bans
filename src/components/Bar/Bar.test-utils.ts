export function recursePermutations<T> (arrOfArrays: T[][], callback: (permutation: T[]) => void, i: number = 0, previousElements: T[] = []) {
  if (i < arrOfArrays.length) {
    const currentElements = arrOfArrays[i]
    for (const element of currentElements) {
      recursePermutations(arrOfArrays, callback, i + 1, previousElements.concat(element))
    }
    if (currentElements.length < 1) {
      recursePermutations(arrOfArrays, callback, i + 1, Array.from(previousElements))
    }
  } else if (previousElements.length > 0) {
    callback(previousElements)
  }
}
