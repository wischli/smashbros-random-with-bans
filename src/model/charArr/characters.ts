export const characters: {
  id: number,
  name: string,
  echo: number[],
}[] = [
    {
    id: 1,
    name: 'Mario',
    echo: [],
  },
  {
    id: 43,
    name: 'Dr. Mario',
    echo: [],
  },
  {
    id: 2,
    name: 'Luigi',
    echo: [],
  },
  {
    id: 3,
    name: 'Peach',
    echo: [3.5],
  },
  {
    id: 3.5,
    name: 'Daisy',
    echo: [3],
  },
  {
    id: 4,
    name: 'Bowser',
    echo: [],
  },
  {
    id: 5,
    name: 'Yoshi',
    echo: [],
  },
  {
    id: 6,
    name: 'Rosalina & Luma',
    echo: [],
  },
  {
    id: 7,
    name: 'Donkey Kong',
    echo: [],
  },
  {
    id: 8,
    name: 'Diddy Kong',
    echo: [],
  },
  {
    id: 9,
    name: 'Link',
    echo: [],
  },
  {
    id: 10,
    name: 'Zelda',
    echo: [],
  },
  {
    id: 11,
    name: 'Sheik',
    echo: [],
  },
  {
    id: 45,
    name: 'Ganondorf',
    echo: [],
  },
  {
    id: 12,
    name: 'Toon Link',
    echo: [],
  },
  {
    id: 12.9,
    name: 'Young Link',
    echo: [],
  },
  {
    id: 13,
    name: 'Samus',
    echo: [13.5],
  },
  {
    id: 13.5,
    name: 'Dark Samus',
    echo: [13],
  },
  {
    id: 14,
    name: 'Zero Suit Samus',
    echo: [],
  },
  {
    id: 15,
    name: 'Pit',
    echo: [15],
  },
  {
    id: 15.5,
    name: 'Dark Pit',
    echo: [15.5],
  },
  {
    id: 16,
    name: 'Palutena',
    echo: [],
  },
  {
    id: 17,
    name: 'Marth',
    echo: [17.5],
  },
  {
    id: 17.5,
    name: 'Lucina',
    echo: [17],
  },
  {
    id: 18,
    name: 'Ike',
    echo: [],
  },
  {
    id: 19,
    name: 'Robin',
    echo: [],
  },
  {
    id: 20,
    name: 'Kirby',
    echo: [],
  },
  {
    id: 21,
    name: 'King Dedede',
    echo: [],
  },
  {
    id: 22,
    name: 'Meta Knight',
    echo: [],
  },
  {
    id: 23,
    name: 'Little Mac',
    echo: [],
  },
  {
    id: 24,
    name: 'Fox',
    echo: [],
  },
  {
    id: 25,
    name: 'Pikachu',
    echo: [],
  },
  {
    id: 25.9,
    name: 'Pichu',
    echo: [],
  },
  {
    id: 26,
    name: 'Pokemon Trainer',
    echo: [],
  },
  {
    id: 27,
    name: 'Lucario',
    echo: [],
  },
  {
    id: 28,
    name: 'Greninja',
    echo: [],
  },
  {
    id: 29,
    name: 'Captain Falcon',
    echo: [],
  },
  {
    id: 30,
    name: 'Villager',
    echo: [],
  },
  {
    id: 31,
    name: 'Olimar',
    echo: [],
  },
  {
    id: 32,
    name: 'Wii Fit Trainer',
    echo: [],
  },
  {
    id: 33,
    name: 'Shulk',
    echo: [],
  },
  {
    id: 34,
    name: 'Pac-Man',
    echo: [],
  },
  {
    id: 35,
    name: 'Mega Man',
    echo: [],
  },
  {
    id: 36,
    name: 'Sonic',
    echo: [],
  },
  {
    id: 38,
    name: 'Ness',
    echo: [],
  },
  {
    id: 39,
    name: 'Falco',
    echo: [],
  },
  {
    id: 40,
    name: 'Wario',
    echo: [],
  },
  {
    id: 41,
    name: 'Snake',
    echo: [],
  },
  {
    id: 42,
    name: 'Ice Climbers',
    echo: [],
  },
  {
    id: 44,
    name: 'R.O.B.',
    echo: [],
  },
  {
    id: 46,
    name: 'Mr. Game & Watch',
    echo: [],
  },
  {
    id: 47,
    name: 'Bowser Jr.',
    echo: [],
  },
  {
    id: 48,
    name: 'Duck Hunt',
    echo: [],
  },
  {
    id: 49,
    name: 'Jigglypuff',
    echo: [],
  },
  {
    id: 50,
    name: 'Mewtwo',
    echo: [],
  },
  {
    id: 51,
    name: 'Lucas',
    echo: [],
  },
  {
    id: 52,
    name: 'Roy',
    echo: [52.5],
  },
  {
    id: 52.5,
    name: 'Chrom',
    echo: [52],
  },
  {
    id: 53,
    name: 'Ryu',
    echo: [53.5],
  },
  {
    id: 53.5,
    name: 'Ken',
    echo: [53],
  },
  {
    id: 54,
    name: 'Cloud',
    echo: [],
  },
  {
    id: 55,
    name: 'Corrin',
    echo: [],
  },
  {
    id: 56,
    name: 'Bayonetta',
    echo: [],
  },
  {
    id: 57,
    name: 'Mii Fighter Brawler',
    echo: [],
  },
  {
    id: 58,
    name: 'Mii Fighter Shooter',
    echo: [],
  },
  {
    id: 59,
    name: 'Mii Fighter Sword',
    echo: [],
  },
  {
    id: 60,
    name: 'Inkling',
    echo: [],
  },
  {
    id: 61,
    name: 'Ridley',
    echo: [],
  },
  {
    id: 62,
    name: 'Simon',
    echo: [62.5],
  },
  {
    id: 62.5,
    name: 'Richter',
    echo: [62],
  },
  {
    id: 63,
    name: 'King K. Rool',
    echo: [],
  },
  {
    id: 64,
    name: 'Isabelle',
    echo: [],
  },
  {
    id: 65,
    name: 'Incineroar',
    echo: [],
  },
  {
    id: 66,
    name: 'Piranha Plant',
    echo: [],
  },
];
