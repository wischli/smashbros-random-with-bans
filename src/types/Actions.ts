export enum Action {
  randomize = 'randomize',
  next = 'next',
  previous = 'previous',
  restore = 'restore',
  echo = 'echo',
  save = 'save',
  reset = 'reset',
  toggleChar = 'toggleChar',
}
