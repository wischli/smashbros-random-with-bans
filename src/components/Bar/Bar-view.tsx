import React, { useContext } from 'react';
import { themeButtonStyle } from '../../layout/themeStyle';
import { saveCharacters } from '../../utils';
import Context, { Icontext } from '../Context';
import { buttonStyle, navStyle } from './Bar-style';

const Bar = () => {
	const {
		state,
		displayCard,
		handleRandomizeClick,
		handleDisplayClick,
		handleEchoClick,
		displayRandomize,
		options
	} = useContext(Context as React.Context<Icontext>);

	// set center button message and action
	let centerBtn: {
		action: Function;
		msg: string;
	};
	if (displayCard) {
		centerBtn = {
			action: () => handleDisplayClick(displayCard),
			msg: 'Close Popup'
		};
	} else if (displayRandomize) {
		centerBtn = {
			action: () => handleDisplayClick(!displayCard),
			msg: 'Show Popup'
		};
	} else {
		centerBtn = {
			action: () => handleRandomizeClick(),
			msg: 'Randomize'
		};
	}

	return (
		<div className="nav" style={navStyle}>
			<button type="button" style={buttonStyle} onClick={() => handleEchoClick()}>
				{options.echo ? 'Hide Echoes' : 'Show Echoes'}
			</button>
			<button type="button" style={{ ...themeButtonStyle, width: '100%' }} onClick={() => centerBtn.action()}>
				{centerBtn.msg}
			</button>
			<button type="button" style={buttonStyle} onClick={() => saveCharacters(state)}>
				{'Save Chars'}
			</button>
		</div>
	);
};
export default Bar;
