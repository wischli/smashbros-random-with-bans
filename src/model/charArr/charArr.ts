import { IChar, IState } from '../../types/Types'
import { characters } from './characters'
import { imigify } from '../../utils'

export const charArr: IChar[] = characters.map(char => ({
  ...char,
  media: imigify(char.name),
  enabled: true,
  played: false,
  display: true,
}))

export const initialCharState: IState = { enabled: charArr.map((_, index) => index), played: [], disabled: [], hidden: [] }
