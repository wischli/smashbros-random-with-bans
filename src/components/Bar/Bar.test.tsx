/* eslint-disable no-undef */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { initialCharState } from '../../model/charArr/charArr';
import { IState } from '../../types/Types';
import Bar from './Bar-view';

configure({ adapter: new Adapter() });

// set default test data
const context = {
	handleRandomizeClick: jest.fn(),
	handleDisplayClick: jest.fn(),
	handleEchoClick: jest.fn()
};
let state: IState;
beforeEach(() => {
	state = { ...initialCharState };
});

const setBtn = ({
	displayCard,
	displayRandomize,
	echo = true
}: {
	displayCard: boolean;
	displayRandomize: boolean;
	echo?: boolean;
}) => {
	const { handleDisplayClick, handleRandomizeClick, handleEchoClick } = context
	return (
		<Bar
			state={state}
			displayCard={displayCard}
			handleDisplayClick={handleDisplayClick}
			handleRandomizeClick={handleRandomizeClick}
			handleEchoClick={handleEchoClick}
			displayRandomize={displayRandomize}
			options={{ echo }}
		/>
	);
};

describe('Testing Bar Component', () => {
	describe('It displays "Close Popup" on center button', () => {
		it('Tests displayCard: true, displayRandomize: true, echo: true', () => {
			const wrapper = shallow(setBtn({ displayCard: true, displayRandomize: true }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Close Popup');
			expect(wrapper).toMatchSnapshot();
		});
		it('Tests displayCard: true, displayRandomize: false, echo: true', () => {
			const wrapper = shallow(setBtn({ displayCard: true, displayRandomize: false }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Close Popup');
			expect(wrapper).toMatchSnapshot();
		});
		it('Tests displayCard: true, displayRandomize: true, echo: false', () => {
			const wrapper = shallow(setBtn({ displayCard: true, displayRandomize: true }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Close Popup');
			expect(wrapper).toMatchSnapshot();
		});
		it('Tests displayCard: true, displayRandomize: false, echo: false', () => {
			const wrapper = shallow(setBtn({ displayCard: true, displayRandomize: false }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Close Popup');
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('It displays "Show Popup" on center button', () => {
		it('Tests displayCard: false, displayRandomize: true, echo: true', () => {
			const wrapper = shallow(setBtn({ displayCard: false, displayRandomize: true }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Show Popup');
			expect(wrapper).toMatchSnapshot();
		});
		it('Tests displayCard: false, displayRandomize: true, echo: false', () => {
			const wrapper = shallow(setBtn({ displayCard: false, displayRandomize: true }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Show Popup');
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('It displays "Randomize" on center button', () => {
		it('Tests displayCard: false, displayRandomize: false, echo: true', () => {
			const wrapper = shallow(setBtn({ displayCard: false, displayRandomize: false }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Randomize');
			expect(wrapper).toMatchSnapshot();
		});
		it('Tests displayCard: false, displayRandomize: false, echo: false', () => {
			const wrapper = shallow(setBtn({ displayCard: false, displayRandomize: false }));
			const centerBtn = wrapper.find('[data-test="centerBtn"]');
			expect(centerBtn).toBeDefined();
			expect(centerBtn.text()).toBe('Randomize');
			expect(wrapper).toMatchSnapshot();
		});
	});
});
