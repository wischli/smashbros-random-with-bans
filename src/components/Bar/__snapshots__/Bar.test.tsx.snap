// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Bar Component It displays "Close Popup" on center button Tests displayCard: true, displayRandomize: false, echo: false 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Close Popup
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Close Popup" on center button Tests displayCard: true, displayRandomize: false, echo: true 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Close Popup
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Close Popup" on center button Tests displayCard: true, displayRandomize: true, echo: false 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Close Popup
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Close Popup" on center button Tests displayCard: true, displayRandomize: true, echo: true 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Close Popup
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Randomize" on center button Tests displayCard: false, displayRandomize: false, echo: false 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Randomize
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Randomize" on center button Tests displayCard: false, displayRandomize: false, echo: true 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Randomize
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Show Popup" on center button Tests displayCard: false, displayRandomize: true, echo: false 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Show Popup
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;

exports[`Testing Bar Component It displays "Show Popup" on center button Tests displayCard: false, displayRandomize: true, echo: true 1`] = `
<div
  className="nav"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#466370",
      "display": "flex",
      "flexWrap": "nowrap",
      "height": 50,
      "justifyContent": "space-between",
      "padding": "10px",
      "position": "fixed",
      "top": 0,
      "width": "calc(100% - 20px)",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Hide Echoes
  </button>
  <button
    data-test="centerBtn"
    onClick={[Function]}
    style={
      Object {
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Show Popup
  </button>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#ddddddb3",
        "border": "2px solid",
        "borderRadius": "5px",
        "fontSize": "15px",
        "fontWeight": 700,
        "height": "50px",
        "justifyContent": "center",
        "letterSpacing": "1px",
        "textAlign": "center",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
    type="button"
  >
    Save Chars
  </button>
</div>
`;
